mito-ai/src/Extensions/AiChat/ChatTaskpane.tsx

```diff
@@ -1343,34 +1358,44 @@ const ChatTaskpane: React.FC<IChatTaskpaneProps> = ({
                         <CTACarousel app={app} />
                     </div>
                 }
-                {displayOptimizedChatHistory.map((displayOptimizedChat, index) => {
-                    return (
-                        <ChatMessage
-                            key={index}
-                            message={displayOptimizedChat.message}
-                            promptType={displayOptimizedChat.promptType}
-                            messageType={displayOptimizedChat.type}
-                            agentResponse={displayOptimizedChat.agentResponse}
-                            codeCellID={displayOptimizedChat.codeCellID}
-                            mitoAIConnectionError={displayOptimizedChat.type === 'connection error'}
-                            mitoAIConnectionErrorType={displayOptimizedChat.mitoAIConnectionErrorType || null}
-                            messageIndex={index}
-                            notebookTracker={notebookTracker}
-                            renderMimeRegistry={renderMimeRegistry}
-                            app={app}
-                            isLastAiMessage={index === lastAIMessagesIndex}
-                            isLastMessage={index === displayOptimizedChatHistory.length - 1}
-                            operatingSystem={operatingSystem}
-                            previewAICode={previewAICodeToActiveCell}
-                            acceptAICode={acceptAICode}
-                            rejectAICode={rejectAICode}
-                            onUpdateMessage={handleUpdateMessage}
-                            contextManager={contextManager}
-                            codeReviewStatus={codeReviewStatus}
-                            setNextSteps={setNextSteps}
-                            agentModeEnabled={agentModeEnabled}
-                        />
-                    )
+                {processedDisplayOptimizedChatHistory.map((displayOptimizedChat, index) => {
+                    if (Array.isArray(displayOptimizedChat)) {
```

Comment:
Can we turn this into a better check? Instead of checking if it's an array, ideally we'd check if the element is of type `GroupedErrorMessages`.